examples/machine-learning/a3-highgpu-8g/a3high-slurm-blueprint.yaml

```diff
@@ -0,0 +1,711 @@
+          apt-get -y update && apt-get -y install google-cloud-cli
+
+      - type: shell
+        destination: install_nccl_openmpi.sh
+        content: |
+          # Install latest NCCL
+          mkdir -p /opt/src
+          cd /opt/src
+          git clone -b v2.25.1-1 https://github.com/NVIDIA/nccl.git
+          cd /opt/src/nccl
+          make -j src.build NVCC_GENCODE="-gencode=arch=compute_90,code=sm_90"
+          make install
+
+          # Install OpenMPI
+          apt-get install -y libopenmpi-dev
+      - type: shell
+        destination: workaround_apt_change.sh
+        content: |
+          #!/bin/bash
+          # this script is no longer necessary on the most recent TCPX A3
+          # images, however it is included for backwards compatibility
+          set -e -o pipefail
+          rm -f /etc/apt/sources.list.d/kubernetes.list
+          apt-get update --allow-releaseinfo-change
+
+      - type: data
+        destination: /var/tmp/slurm_vars.json
+        content: |
+          {
+            "reboot": false,
+            "install_cuda": false,
+            "install_gcsfuse": true,
+            "install_lustre": false,
+            "install_ompi": true,
+            "monitoring_agent": "cloud-ops",
+            "nvidia_version": "latest",
+            "allow_kernel_upgrades": true, # tcpx kernel is held and gcp kernel is removed as the last task
+            "install_nvidia_repo": false
+          }
+
+      - type: shell
+        destination: install_slurm.sh
+        content: |
+          #!/bin/bash
+          set -e -o pipefail
+          ansible-galaxy role install googlecloudplatform.google_cloud_ops_agents
+          ansible-pull \
+              -U https://github.com/GoogleCloudPlatform/slurm-gcp -C 6.10.7 \
+              -i localhost, --limit localhost --connection=local \
+              -e @/var/tmp/slurm_vars.json \
+              ansible/playbook.yml
+
+      - type: data
+        destination: /etc/security/limits.d/99-unlimited.conf
+        content: |
+          * - memlock unlimited
+          * - nproc unlimited
+          * - stack unlimited
+          * - nofile 1048576
+          * - cpu unlimited
+          * - rtprio unlimited
+
+      - type: data
+        destination: /etc/enroot/enroot.conf
+        content: |
+          ENROOT_RUNTIME_PATH    /mnt/localssd/${UID}/enroot/runtime
+          ENROOT_CACHE_PATH      /mnt/localssd/${UID}/enroot/cache
+          ENROOT_DATA_PATH       /mnt/localssd/${UID}/enroot/data
+          ENROOT_TEMP_PATH       /mnt/localssd/${UID}/enroot
+      - type: shell
+        destination: install_mdadm.sh
+        content: |
+          #!/bin/bash
+          # this script ensures that the mdadm package is already present when
+          # compute nodes boot and use MDADM to RAID local SSD disks
+          apt-get update
+          apt-get install mdadm --no-install-recommends --assume-yes
+      # Add gpu_rxq configs
+      - type: data
```

Comment:
wondering why this is not grouped with other tcpx related runners.

Reply:
We could. I placed it following slurm installation as its a prerequisite for the slurm prolog script.